pkg/cmd/certregenerationcontroller/certregenerationcontroller.go

```diff
@@ -0,0 +1,208 @@
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/operatorclient"
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/targetconfigcontroller"
+	"github.com/openshift/library-go/pkg/operator/certrotation"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/genericoperatorclient"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+)
+
+type Options struct {
+	Workers        int
+	KubeconfigPath string
+	TlsServerName  string
+
+	restConfig   *restclient.Config
+	kubeClient   kubernetes.Interface
+	configClient configeversionedclient.Interface
+}
+
+func NewCertRegenerationControllerCommand() *cobra.Command {
+	o := &Options{
+		Workers:        2,
+		KubeconfigPath: "",
+		TlsServerName:  "",
+	}
+
+	cmd := &cobra.Command{
+		Use: "cert-regeneration-controller",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			err := o.Validate()
+			if err != nil {
+				return err
+			}
+
+			err = o.Complete()
+			if err != nil {
+				return err
+			}
+
+			err = o.Run()
+			if err != nil {
+				return err
+			}
+
+			return nil
+		},
+
+		SilenceErrors: true,
+		SilenceUsage:  true,
+	}
+
+	cmd.PersistentFlags().IntVarP(&o.Workers, "workers", "", o.Workers, "Number of workers to run")
+	cmd.PersistentFlags().StringVarP(&o.KubeconfigPath, "kubeconfig", "", o.KubeconfigPath, "Path to the kubeconfig file")
+	cmd.PersistentFlags().StringVarP(&o.TlsServerName, "tls-server-name", "", o.TlsServerName, "The SNI hostname to set for the server in kubeconfig")
+
+	return cmd
+}
+
+func (o *Options) Validate() error {
+	if o.Workers <= 0 {
+		return fmt.Errorf("number of workers must be > 0")
+	}
+
+	return nil
+}
+
+func (o *Options) Complete() error {
+	var err error
+
+	if len(o.KubeconfigPath) != 0 {
+		klog.V(1).Infof("Using kubeconfig %q.", o.KubeconfigPath)
+		o.restConfig, err = clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+			&clientcmd.ClientConfigLoadingRules{ExplicitPath: o.KubeconfigPath}, &clientcmd.ConfigOverrides{}).ClientConfig()
+		if err != nil {
+			return fmt.Errorf("can't create config from kubeConfigPath %q: %v", o.KubeconfigPath, err)
+		}
+	} else {
+		klog.V(1).Infof("No kubeconfig specified, using InClusterConfig.")
+		o.restConfig, err = restclient.InClusterConfig()
+		if err != nil {
+			return fmt.Errorf("can't create InClusterConfig: %v", err)
+		}
+	}
+
+	if len(o.TlsServerName) != 0 {
+		o.restConfig.TLSClientConfig.ServerName = o.TlsServerName
+	}
+
+	o.kubeClient, err = kubernetes.NewForConfig(o.restConfig)
+	if err != nil {
+		return fmt.Errorf("can't build kubernetes clientset: %v", err)
+	}
+
+	o.configClient, err = configeversionedclient.NewForConfig(o.restConfig)
+	if err != nil {
+		return fmt.Errorf("failed to create config client: %v", err)
+	}
+
+	return nil
+}
+
+func (o *Options) Run() error {
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
@deads2k do we want leader election between recovery controllers or just keeping the premise of acting only if the cluster has broken certs?

Reply:
leader election added